### ISSUE TITLE
Cache path doesn't show cached file

### ISSUE THREAD

**Issue by bangedorrunt** [rktjmp/hotpot.nvim#83]
@rktjmp I follow the cookbook to quickly open a lua cache of current fnl file
```fennel
(noremap n nowait "<leader>hc"
           #(let [{: cache-path-for-fnl-file} (require :hotpot.api.cache)]
              (match (cache-path-for-fnl-file (vim.fn.expand :%:p))
                path (vim.cmd (.. ":new " path))
                nil (vim.api.nvim_echo [["No cache file for current file" :WarningMsg]] true {}))))
```
However, it won't found the cache file, eventhough it's available from `lua` folder. Am I missing sth here?
Note that I'm using `aot` feature also. Is it the cause?

fnl files:
<img width="936" alt="Screen Shot 2022-09-13 at 12 40 56 pm" src="https://user-images.githubusercontent.com/1514823/189795613-4eca927a-df66-4223-a5e8-64652f9795c5.png">

lua files:
<img width="939" alt="Screen Shot 2022-09-13 at 12 42 15 pm" src="https://user-images.githubusercontent.com/1514823/189795795-9bfc7c02-efe1-4678-9416-ccf2aceacdea.png">

Use with `Telescope`, there is only 2 files available:
<img width="1128" alt="Screen Shot 2022-09-13 at 12 45 05 pm" src="https://user-images.githubusercontent.com/1514823/189796135-51ab0668-f3bb-4386-b542-5b1e8bd47207.png">

**Comment by rktjmp** [comment]
The cache api only knows about *cached* files, that are in the compile on demand cache dir. AOT files are not tracked as they could end up anywhere. I would match on the path and either direct to your AOT dir or fall back to the cache dir, or maybe you can re-use your AOT pattern matching functions to generate the path.

**Comment by rktjmp** [comment]
Basically the cache is only used for on-demand files, because we need "somewhere" to put the lua output, so they get put in a cache folder for loading next time. (Technically since 2.0 they are now also ... cached ... as bytecode in index.bin but the lua files are retained for convenience.)

When you do AOT, you're kind of walking out on your own, it's assumed that you are putting the files in an actual "loadable" location for example - eg under some `lua/` folder, so they never go in the cache folder. The `api.cache` functions are really just shortcuts to manipulate things in `~/.cache/nvim/hotpot`.

So basically `cache = on-demand = hotpot-managed` and `aot = user-managed`.
